src/Evolution/Executables/ScalarWave/EvolveScalarWave.hpp

```diff
@@ -35,37 +35,40 @@
 
 template <size_t Dim>
 struct EvolutionMetavars {
+ private:
+  using temporal_id_tag = Tags::TimeId;
+
+ public:
   // Customization/"input options" to simulation
   using system = ScalarWave::System<Dim>;
-  using temporal_id = Tags::TimeId;
   using analytic_solution_tag =
       CacheTags::AnalyticSolution<ScalarWave::Solutions::PlaneWave<Dim>>;
-  using normal_dot_numerical_flux =
+  using numerical_flux_tag =
       CacheTags::NumericalFluxParams<ScalarWave::UpwindFlux<Dim>>;
   // A tmpl::list of tags to be added to the ConstGlobalCache by the
   // metavariables
   using const_global_cache_tag_list = tmpl::list<analytic_solution_tag>;
   using domain_creator_tag = OptionTags::DomainCreator<Dim, Frame::Inertial>;
 
-  using component_list = tmpl::list<DgElementArray<
-      EvolutionMetavars,
-      tmpl::list<Actions::ComputeVolumeDuDt<Dim>,
-                 dg::Actions::ComputeNonconservativeBoundaryFluxes,
-                 dg::Actions::SendDataForFluxes<EvolutionMetavars>,
-                 dg::Actions::ReceiveDataForFluxes<EvolutionMetavars>,
-                 dg::Actions::ApplyBoundaryFluxesGlobalTimeStepping,
-                 Actions::UpdateU, Actions::AdvanceTime, Actions::FinalTime>>>;
+  using evolve_algorithm_actions = tmpl::list<
+      Actions::ComputeVolumeDuDt<Dim>,
+      dg::Actions::ComputeNonconservativeBoundaryFluxes<
+          Dim, ScalarWave::ComputeNormalDotFluxes<Dim>>,
+      dg::Actions::SendDataForFluxes<temporal_id_tag, numerical_flux_tag>,
+      dg::Actions::ReceiveDataForFluxes<Dim, temporal_id_tag,
+                                        numerical_flux_tag>,
+      dg::Actions::ApplyBoundaryFluxesGlobalTimeStepping<temporal_id_tag,
+                                                         numerical_flux_tag>,
+      Actions::UpdateU, Actions::AdvanceTime, Actions::FinalTime>;
+  using component_list =
+      tmpl::list<DgElementArray<EvolutionMetavars, evolve_algorithm_actions>>;
 
   static constexpr OptionString help{
       "Evolve a Scalar Wave in Dim spatial dimension.\n\n"
       "The analytic solution is: PlaneWave\n"
       "The numerical flux is:    UpwindFlux\n"};
 
-  enum class Phase {
-    Initialization,
-    Evolve,
-    Exit
-  };
+  enum class Phase { Initialization, Evolve, Exit };
```

Comment:
again, try to avoid these types of changes in commits that require us to understand the code in detail.